tests/server/test_handlers.py

```diff
@@ -641,3 +643,39 @@ def test_delete_model_version_tag(mock_get_request_message, mock_model_registry_
     _delete_model_version_tag()
     _, args = mock_model_registry_store.delete_model_version_tag.call_args
     assert args == {"name": name, "version": version, "key": key}
+
+
+@pytest.mark.skipif(is_windows(), reason="This test fails on Windows")
+@pytest.mark.parametrize(
+    ("file_path", "expected_mime_type"),
+    [
+        ("/a/b/c.txt", "text/plain"),
+        ("c.txt", "text/plain"),
+        ("c.pkl", "application/octet-stream"),
+        ("/a/b/c.pkl", "application/octet-stream"),
+        ("/a/b/c.png", "image/png"),
+        ("/a/b/c.pdf", "application/pdf"),
+        ("/a/b/MLmodel", "text/plain"),
+        ("/a/b/mlproject", "text/plain"),
+    ],
+)
+def test_guess_mime_type(file_path, expected_mime_type):
+    assert _guess_mime_type(file_path) == expected_mime_type
+
+
+@pytest.mark.skipif(not is_windows(), reason="This test only passes on Windows")
+@pytest.mark.parametrize(
+    ("file_path", "expected_mime_type"),
+    [
+        ("C:\\a\\b\\c.txt", "text/plain"),
+        ("c.txt", "text/plain"),
+        ("c.pkl", "application/octet-stream"),
+        ("C:\\a\\b\\c.pkl", "application/octet-stream"),
+        ("C:\\a\\b\\c.png", "image/png"),
+        ("C:\\a\\b\\c.pdf", "application/pdf"),
+        ("C:\\a\\b\\MLmodel", "text/plain"),
+        ("C:\\a\\b\\mlproject", "text/plain"),
+    ],
+)
+def test_guess_mime_type_on_windows(file_path, expected_mime_type):
```

Comment:
```suggestion
@pytest.mark.parametrize(
    ("file_path", "expected_mime_type"),
    [
        ("/a/b/c.txt", "text/plain"),
        ("c.txt", "text/plain"),
        ("c.pkl", "application/octet-stream"),
        ("/a/b/c.pkl", "application/octet-stream"),
        ("/a/b/c.png", "image/png"),
        ("/a/b/c.pdf", "application/pdf"),
        ("/a/b/MLmodel", "text/plain"),
        ("/a/b/mlproject", "text/plain"),
        ("C:\\a\\b\\c.txt", "text/plain"),
        ("c.txt", "text/plain"),
        ("c.pkl", "application/octet-stream"),
        ("C:\\a\\b\\c.pkl", "application/octet-stream"),
        ("C:\\a\\b\\c.png", "image/png"),
        ("C:\\a\\b\\c.pdf", "application/pdf"),
        ("C:\\a\\b\\MLmodel", "text/plain"),
        ("C:\\a\\b\\mlproject", "text/plain"),
    ],
)
def test_guess_mime_type(file_path, expected_mime_type):
```

I think we can merge these tests.

Reply:
I tested this with the above change `filename = pathlib.Path(file_path).name`, on unix systems that returns the entire path as name.
input: `pathlib.Path("C:\\a\\b\\MLmodel").name`
output: `C:\\a\\b\\MLmodel`

I think we need to separate those tests by OS here to assert that the artifact path resolves to a correct mime type.

Reply:
Got it, thanks for checking!